feat: add HTTP/2 support via environment variable

aramikuto opened the pull request:
## In this PR:

- New feature (non-breaking change that adds functionality)

Some cloud platforms impose request size limits on HTTP/1 connections, which can impact larger cache artifacts. HTTP/2 support removes these limitations, enabling more efficient caching for larger projects.

I have tested the deployment to Cloud Run with HTTP/2 end-to-end enabled, and verified that the remote cache works with this configuration.

### Checklist:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/ducktors/turborepo-remote-cache/pulls) for the same update/change?
<!-- You can erase any parts of this template not applicable to your Pull Request. -->
* [x] Have you linted your code locally with `pnpm lint` before submission?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully built locally with `pnpm build`?
* [x] Have you successfully tested locally with `pnpm test`?
* [x] Have you committed using [Conventional Commits](https://github.com/ducktors/turborepo-remote-cache#how-to-commit)?

aramikuto commented on src/app.ts:
```diff
@@ -13,6 +13,7 @@ export function createApp(
 ): FastifyInstance {
   const fastifyOptions: FastifyServerOptions = {
     ...options,
+    ...(process.env.HTTP2 === 'true' ? { http2: true } : {}),
```
We pass the **http2** flag configuration conditionally here because the Fastify package types do not support **false** as a value:
https://github.com/fastify/fastify/issues/5682

matteovivona replied:
@fox1t fyi

fox1t replied:
Is this enough to have only the plain text support? https://fastify.dev/docs/latest/Reference/HTTP2/#plain-or-insecure

aramikuto replied:
@fox1t Thank you for the review!

For Cloud Run, [only h2c is supported](https://cloud.google.com/run/docs/configuring/http2#before_you_configure). On AWS Lambda and Azure Functions, the allowed payload size isn't affected by the HTTP version. I haven't found other serverless platforms where HTTP/2 support would be beneficial. Additionally, [handling encryption with Node.js](https://fastify.dev/docs/latest/Guides/Recommendations/#use-a-reverse-proxy) isn't recommended due to performance issues, so plain text should be sufficient.

fox1t commented:
@aramikuto can we find a way to test this?

aramikuto commented:
@fox1t To test it, I've deployed it to Cloud Run. Do you have access to any Google Cloud projects where you can create a new Cloud Run service? If so, I can prepare instructions for deploying it there. If that's not an option, I'll explore ways to test it locally instead.

matteovivona commented:
> @fox1t To test it, I've deployed it to Cloud Run. Do you have access to any Google Cloud projects where you can create a new Cloud Run service? If so, I can prepare instructions for deploying it there. If that's not an option, I'll explore ways to test it locally instead.

could you create something like this? https://github.com/ducktors/turborepo-remote-cache/blob/main/docs/running-in-lambda.md 

so I'll try it directly on cloud run

aramikuto commented:
I'll add the deployment instructions to this PR today.

aramikuto commented:
@matteovivona sorry for the delay. I've updated the documentation to include instructions for deploying to Cloud Run. Since plaintext HTTP/2 isn't useful outside Cloud Run deployments, I've removed the separate HTTP/2 documentation page and integrated its relevant content into the Cloud Run instructions.

To test the unreleased changes, you'll need to build the Docker container locally, [push it to the Cloud Registry](https://cloud.google.com/artifact-registry/docs/docker/store-docker-container-images) and use this image when deploying your Cloud Run service. The remaining deployment steps should be covered by the added [deployment guid](https://github.com/ducktors/turborepo-remote-cache/blob/b42a2c87fb2a36e10c7407807308331a68fd1e79/docs/running-in-cloud-run.md). Let me know if you have any questions!

neilenns commented on src/app.ts:
```diff
@@ -13,6 +13,7 @@ export function createApp(
 ): FastifyInstance {
   const fastifyOptions: FastifyServerOptions = {
     ...options,
+    ...(process.env.HTTP2 === 'true' ? { http2: true } : {}),
```
Should HTTP2 be added to `env.ts` instead and accessed via `env.get().HTTP2`?

matteovivona commented:
@all-contributors add @aramikuto on code
